src/client/ClientGameEngine.cpp

```diff
@@ -17,40 +20,112 @@ void ClientGameEngine::handleInformation(int x, int y)
 }
 
 void ClientGameEngine::renderGame() 
 {
-    std::thread t(&ClientGameEngine::startGameWindow, this);
-    t.detach();
-
-    long lastUpdateTimer = clientGame.getCurrentTime();
-    struct stat file_stat;
-    std::string file_path = RESOURCES_PATH "/img/map/files.json";
-    if (stat(file_path.c_str(), &file_stat) == -1) {
-        std::cerr << "Erreur lors de l'obtention des informations sur le fichier " << file_path << '\n';
-    }
-    time_t last_modified = file_stat.st_mtime;
+    clientWindow = std::make_shared<sf::RenderWindow>();
+    clientWindow->create(sf::VideoMode(WINDOW_LENGTH, WINDOW_WIDTH), "Civilization VII", sf::Style::Close);
+    clientWindow->setPosition(sf::Vector2i(0, 0));
+
+    while (true) {
 
-    while(1){
+        std::unique_lock<std::mutex> lockGlobalWhile(mutexRunningEngine);
+        if (!runningWindow) {
+            lockGlobalWhile.unlock();
+            clientWindow->close();
+            return;
+        }
+        lockGlobalWhile.unlock();
         
-        if (clientGame.getCurrentTime() - lastUpdateTimer > REFRESH_ELEMENT ){
+        playGame();
+        playMenu();
+    }
+}
+
+void ClientGameEngine::playGame() 
+{
+    std::unique_lock<std::mutex> lockIf(mutexRunningEngine);
+    if (runningWindow == 2) {
+        lockIf.unlock();
+
+        std::thread t(&ClientGameEngine::startGameWindow, this);
+
+        long lastUpdateTimer = clientGame.getCurrentTime();
+        struct stat file_stat;
+        std::string file_path = RESOURCES_PATH "/img/map/files.json";
+        if (stat(file_path.c_str(), &file_stat) == -1) {
+            std::cerr << "Erreur lors de l'obtention des informations sur le fichier " << file_path << '\n';
+        }
+        time_t last_modified = file_stat.st_mtime;
 
-            if (stat(file_path.c_str(), &file_stat) == -1) {
-                std::cerr << "Erreur lors de l'obtention des informations sur le fichier " << file_path << '\n';
+        while (true) {
+
+            std::unique_lock<std::mutex> lockWhile(mutexRunningEngine);
+            if (runningWindow != 2) {
+                lockWhile.unlock();
+                t.join();
+                break;
             }
+            lockWhile.unlock();
+
+            if (clientGame.getCurrentTime() - lastUpdateTimer > REFRESH_ELEMENT ){
 
-            time_t modified = file_stat.st_mtime;
-            lastUpdateTimer = clientGame.getCurrentTime();
+                if (stat(file_path.c_str(), &file_stat) == -1) {
+                    std::cerr << "Erreur lors de l'obtention des informations sur le fichier " << file_path << '\n';
+                }
 
-            if (modified != last_modified){
-                last_modified = modified;
-                clientGame.updateElementTexture();
+                time_t modified = file_stat.st_mtime;
+                lastUpdateTimer = clientGame.getCurrentTime();
+
+                if (modified != last_modified){
```

Comment:
Why do we need `last_modified `?

Reply:
to check if it steel the same time for the last modification.

Reply:
It can change few times, but it will be different with the future way to update element so it doesn't care